### ISSUE TITLE
[Bug] Double Free / Negative Ref Count when using cachegen serializer on AMD ROCm

### ISSUE THREAD

**Issue by Young-TW** [LMCache/LMCache#2409]
**Label**
bug, rocm

**Describe the bug**
When running LMCache with the `cachegen` serializer on an AMD ROCm environment (MI300X), the Consumer triggers a "Double Free" warning and eventual instability during the decoding phase.

The system logs repetitive warnings about negative reference counts for `MemoryObj`, indicating a conflict in memory ownership management between the `cachegen` C++ extension and PyTorch/vLLM on ROCm.

**To Reproduce**

1. **Environment**:
* **Hardware**: AMD MI300X (8x)
* **Docker Image**: `rocm/vllm-dev:nightly_main_20260112`
* **LMCache Config**: Set `remote_serde: "cachegen"` in `lmcache_config.yaml`.

2. **Steps**:
* Start Producer and Consumer on the ROCm node.
* Send a prefill request to the Producer (to generate KV cache).
* Send the same request to the Consumer (to trigger retrieval and decoding).

3. **Observation**:
* The Consumer successfully retrieves the cache (Metadata is correct, LMCache hit tokens > 0).
* During the decoding phase (specifically in `cachegen_decoder`), the following warning floods the logs.

**Expected behavior**
The `cachegen` serializer should handle tensor memory ownership correctly on ROCm without triggering double free errors, similar to how it behaves on CUDA. The decoding process should complete without negative reference count warnings.

**Logs**

```text
lmcache_consumer | (EngineCore_DP0 pid=585) [2026-01-14 02:16:20,120] LMCache WARNING: Ref count of MemoryObj -1 is negative: -1. Double free occurred somewhere. Setting ref count back to 0 as a hack but please find the bug. (memory_management.py:475:lmcache.v1.memory_management)
lmcache_consumer | (EngineCore_DP0 pid=585) [2026-01-14 02:16:20,120] LMCache WARNING: Ref count of MemoryObj -1 is negative: -1. Double free occurred somewhere. Setting ref count back to 0 as a hack but please find the bug. (memory_management.py:475:lmcache.v1.memory_management)

```

**Desktop (please complete the following information):**

* **OS**: Linux (Ubuntu Container)
* **Hardware**: AMD Instinct MI300X
* **Environment**: Docker (`rocm/vllm-dev:nightly_main_20260112`)

**Additional context**

1. **Instability / Undefined Behavior**: I observed successful runs in previous attempts (possibly with slightly older vLLM nightly builds or different memory layouts), but the issue has become persistent in the current environment (`rocm/vllm-dev:nightly_main_20260112`). This suggests a race condition or undefined behavior in memory management that is triggered more frequently with recent vLLM changes.
2. **Control Group**: Switching to `remote_serde: "naive"` (pickle) completely resolves the issue, confirming that the bug is specific to the `cachegen` backend's interaction with ROCm memory.
3. **Temporary Workaround**: I found that forcing a deep copy in `lmcache/v1/storage_backend/naive_serde/cachegen_decoder.py` bypasses the crash. By detaching the tensor from the C++ extension's memory management, the double free is avoided:

```python
# Workaround in cachegen_decoder.py
def decode(self, buffer, shape, dtype):
    # ...
    raw_tensor = cachegen_basics.decode(buffer, shape, dtype)
    
    # FORCE DEEP COPY: This bypasses the double free error
    safe_tensor = raw_tensor.clone().detach()
    
    # Optional: manual delete to test if it crashes immediately
    del raw_tensor
    
    return safe_tensor
```
